File: wk/MatlabScripts/Fourier1D.m
%====================================================
%
% Compute 1D Fourier Transform of Function F(1:nt,1:nx)
% along the coordinate X
%
%   Input: F(1:nt,1:nx),  !!!  "nx" has to be odd  !!!
%	   x(1:nx)
%
%   Output: TFF(1:nt,1:nx)
%	    kx(1:nx)
%
%====================================================

function [TFF,kx] = Fourier1D(F0,x0)

nx0 = length(x0);
nx  = 2*fix(nx0/2);
hnx = nx/2;
x   = x0(1:nx);
F   = F0(:,1:nx);

%--------------------------------------------
% Creation of the vector in the Fourier space
% kx(1:nx)

Lx           = x(nx) - x(1);
dx           = x(2) - x(1);
dkx          = 2*pi./(Lx+dx);
kx(1:hnx)    = -(hnx:-1:1) * dkx;
kx(hnx+1:nx) = (0:hnx-1) * dkx;

%--------------------------------------------
% Fourier transformation using Matlab fft
n1  = size(F,1);
TFF = zeros(n1,nx);

var = fft(F')'/nx;
TFF(:,1:hnx)    = var(:,hnx+1:nx);
TFF(:,hnx+1:nx) = var(:,1:hnx);

File: post_process/PythonScripts/fourier1D.py
import numpy as np

def fourier_1d(F0, x0):
    nx0 = len(x0)
    nx = 2 * (nx0 // 2)
    hnx = nx // 2
    x = x0[:nx]
    F = F0[:nx]

    # Creation of the vector in the Fourier space kx(1:nx)
    Lx = x[-1] - x[0]
    dx = x[1] - x[0]
    dkx = 2 * np.pi / (Lx + dx)
    kx = np.zeros(nx)
    kx[:hnx] = -(np.arange(hnx, 0, -1)) * dkx
    kx[hnx:nx] = np.arange(0, hnx) * dkx

    # Fourier transformation using numpy's fft
    TFF = np.zeros_like(F, dtype=np.complex128)
    var = np.fft.fft(F) / nx
    TFF[:hnx] = var[hnx:nx]
    TFF[hnx:nx] = var[:hnx]

    return TFF, kx